ksp_plugin/plugin.cpp

```diff
@@ -544,36 +559,67 @@ void Plugin::SetPartApparentDegreesOfFreedom(
 DegreesOfFreedom<World> Plugin::GetPartActualDegreesOfFreedom(
     PartId const part_id,
     PartId const part_at_origin) const {
-  auto const world_origin = FindOrDie(part_id_to_vessel_, part_at_origin)->
-                                part(part_at_origin)->
-                                degrees_of_freedom();
-  RigidMotion<Barycentric, World> barycentric_to_world{
-      RigidTransformation<Barycentric, World>{
+  enum class LocalTag { tag };
+  using MainBodyCentred =
+      geometry::Frame<LocalTag, LocalTag::tag, /*frame_is_inertial=*/false>;
```

Comment:
This frame already exists somewhere else in `plugin.cpp`.  Let's define it properly in `frames.hpp`.